src/client/particles.cpp

```diff
@@ -299,16 +301,21 @@ ParticleSpawner::ParticleSpawner(
 }
 
 void ParticleSpawner::spawnParticle(ClientEnvironment *env, float radius,
-	bool is_attached, const v3f &attached_pos, float attached_yaw)
+	const core::matrix4 *attached_absolute_pos_rot_matrix)
 {
 	v3f ppos = m_player->getPosition() / BS;
 	v3f pos = random_v3f(m_minpos, m_maxpos);
 
 	// Need to apply this first or the following check
 	// will be wrong for attached spawners
-	if (is_attached) {
-		pos.rotateXZBy(attached_yaw);
-		pos += attached_pos;
+	if (attached_absolute_pos_rot_matrix) {
+		pos *= BS;
+		attached_absolute_pos_rot_matrix->transformVect(pos);
+		pos /= BS;
+		v3s16 camera_offset = m_particlemanager->m_env->getCameraOffset();
+		pos.X += camera_offset.X;
+		pos.Y += camera_offset.Y;
+		pos.Z += camera_offset.Z;
```

Comment:
Why not `pos += m_particlemanager->m_env->getCameraOffset();`?

Reply:
Will do

Reply:
This causes a build error, as pos is a `v3f`, while `getCameraOffset` returns a `v3s16`. Probably why pgimeno manually added the components in the first place.